tools/system_libs.py

```diff
@@ -770,11 +775,16 @@ def do_create():
   # Wrap libraries in --whole-archive, as needed.  We need to do this last
   # since otherwise the abort sorting won't make sense.
   ret = []
-  for name, need_whole_archive in libs_to_link:
-    if need_whole_archive:
-      ret += ['--whole-archive', name, '--no-whole-archive']
-    else:
-      ret.append(name)
+  if force_all or shared.Settings.LINKABLE:
```

Comment:
instead of this part of the change, how about setting `need_whole_archive` on 717/722 to True if linkable?

and would that avoid the need for `force_all`? or maybe I don't understand why that part is needed.

Reply:
I could do, but the addition of force_all also tidies up the command line, so you don't end up with an pair of  `--whole-archive,--no-whole-archive` around every element in the list. 

Reduces the noise when looking at the linker command line.

Reply:
I actually ended up solving this a different way for the my dynamic linker PR since the --whole-archive thing needs to also apply to user libraries I just wrapped the whole lld command line this flag.  

So I'm going to revert the LINKABLE part here I think.

Reply:
I split out part of this change: https://github.com/emscripten-core/emscripten/pull/8426

Reply:
I see, sounds good. lgtm.